pkg/traces/automaticloggingprocessor/automaticloggingprocessor.go

```diff
@@ -183,8 +183,8 @@ func (p *automaticLoggingProcessor) Capabilities() consumer.Capabilities {
 
 // Start is invoked during service startup.
 func (p *automaticLoggingProcessor) Start(ctx context.Context, _ component.Host) error {
-	logs := ctx.Value(contextkeys.Logs).(*logs.Logs)
-	if logs == nil {
+	logs, ok := ctx.Value(contextkeys.Logs).(*logs.Logs)
+	if !ok {
```

Comment:
Just in case you don't know: if you don't care about the value of `ok` you can also do `logs, _ := ctx.Value(contextkeys.Logs).(*logs.Logs)` here. Sometimes helpful :) 

This is good as-is though!

Reply:
Oh TIL, thanks!